packages/compat/src/resolver.ts

```diff
@@ -731,6 +731,11 @@ export default class CompatResolver implements Resolver {
         from
       );
     }
+
+    if (builtInComponents.includes(component.path)) {
```

Comment:
Hmm, I just realized that early returning here will prevent people from "shadowing" built-in components by creating components with the same name in their project or addon. It doesn't seem like a good thing to do and I'm not sure if it's officially supported but it does seem to work (tested in Ember twiddle).

Is that something that needs to work under Embroider?

Reply:
as I understand, we could check it using 'tryComponent' method